index.bs

```diff
@@ -739,50 +945,132 @@ interface MLOperand {};
 
 #### Creating {{MLOperand}} #### {#api-mloperand-create}
 The {{MLOperand}} objects are created by the methods of {{MLGraphBuilder}}, internally using the following algorithms.
 
-To <dfn>create MLOperand</dfn> given |builder| and |desc|, run the following steps:
-<div class=algorithm-steps>
-    1. If |builder| is not an instance of {{MLGraphBuilder}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
-    1. If |desc| is not an [=object=] that [=implements=] {{MLOperandDescriptor}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
+<details open>
+  <summary>
+    To <dfn>create MLOperand</dfn> given |builder| and |desc|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |builder| is {{MLGraphBuilder}}.
+    1. If |desc| is not an [=object=] that [=implements=] {{MLOperandDescriptor}}, then [=exception/throw=] a {{TypeError}} and stop.
     1. Let |operand| be a new [=object=].
     1. Set |operand|.{{MLOperand/[[builder]]}} to |builder|.
     1. Set |operand|.{{MLOperand/[[descriptor]]}} to |desc|.
     1. Return |operand|.
-</div>
-
-To <dfn>check dimensions</dfn> given |dimensions| and |type|, run the following steps:
-<div class=algorithm-steps>
+  </div>
+</details>
+
+<details open>
+  <summary>
+    To <dfn>copy MLOperand</dfn> given |operand|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |operand| is {{MLOperand}}.
+    1. Let |result| be a new [=object=].
+    1. Set |result|.{{MLOperand/[[builder]]}} to |operand|.{{MLOperand/[[builder]]}}.
+    1. Set |result|.{{MLOperand/[[descriptor]]}} to |operand|.{{MLOperand/[[descriptor]]}}.
+    1. If |operand|.{{MLOperand/[[name]]}} [=map/exists=], then set |result|.{{MLOperand/[[name]]}} to |operand|.{{MLOperand/[[name]]}}.
+    1. Return |result|.
+  </div>
+</details>
+
+<details open>
+  <summary>
+    To <dfn>check dimensions</dfn> given |dimensions| and |type|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
     1. If |dimensions| is not an array of positive numbers, return `false`;
     1. If |dimensions|.length is 0, return `false`.
     1. If |dimensions|.length is too large to be supported by the implementation, return `false`.
     1. If any element of |dimensions| is not a positive number, or it is too large to be supported by the implementation given |type|, return `false`.
     1. Return `true`.
-</div>
+  </div>
+</details>
+
+<details open>
+  <summary>
+    To <dfn for="MLOperand">validate MLOperand</dfn> given |operand| and |builder|, run the following steps:
+  </summary>
+  <div class=algorithm-steps>
+    1. [=Assert=]: the type of |operand|.{{MLOperand/[[builder]]}} is {{MLGraphBuilder}}.
+    1. If |builder| [=map/exists=]] and is not equal to |operand|.{{MLOperand/[[builder]]}}, return `false`.
```

Comment:
Stray `]` after exists. But also since _builder_ is explicitly passed I don't think the "exists" check makes any sense?